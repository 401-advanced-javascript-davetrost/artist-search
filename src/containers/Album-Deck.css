.AlbumDeck {

  h1 {
    color: #e97e31;
    font-family: 'Bowlby One SC', cursive;
    font-size: 35px;
    letter-spacing: .2em;
    line-height: 1.1em;
    margin: .2em;
    margin-top: .5em;
    text-align: center;
    
    span {
      font-size: 20px;
    }
  }

  h2 {
    color: black;
    font-family: 'Questrial', sans-serif;
    line-height: 1.1em;
    margin-bottom: 5px;
    font-size: 20pt;
  }

  section {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: auto;
  
    button {
      width: 15%;
      height: 30px;
      margin-top: 15%;
      background: #e97e31;
      color: whitesmoke;
      font-size: 1.2em;
      text-transform: uppercase;
      font-family: 'Questrial', sans-serif;
      font-weight: 700;
      &:hover {
        cursor: pointer;
        background-color: rgba(233, 126, 49, 0.75);
      }
    }
  
    ul {
      margin-bottom: 100px;
      width: 40vw;
      display: flex;
      flex-wrap: wrap;
      justify-content: space-evenly;
      
      div {
        width: 15vw;
        margin: 0;
        padding: 10px;
        text-align: center;
        margin-bottom: 35px;

        Img {
          width: 15em;
        }

        h2 {
          text-decoration: none;
          color: black;
          font-family: 'Questrial', sans-serif;
          font-size: 1.3em;
          line-height: 1.1em;
          margin: 0
        }
    
      }

    }
    
    a, h2 {
      text-decoration: none;
      color: black;
      font-family: 'Questrial', sans-serif;
      line-height: 1.1em;
    }
  
    a:hover {
      color: #459DB4;
      cursor: pointer;
      text-decoration: none;
    }
  
  }
}