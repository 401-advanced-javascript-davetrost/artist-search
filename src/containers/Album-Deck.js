import React, { memo, useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import AlbumCard from '../components/Album-Card';
import { getAlbums } from '../services/artist-api';
import styles from './Album-Deck.css';

function AlbumDeck() {
  const [albums, setAlbums] = useState([]);
  const [page, setPage] = useState(1);
  const { id, artist } = useParams();

  useEffect(() => {
    getReleases(page);
  }, [page]);

  const handlePageChange = page => {
    const newPage = Math.max(page, 1);
    getReleases(newPage);
    setPage(newPage);
  };

  const getReleases = page => {
    getAlbums(id, page)
      .then(({ releases }) => {
        setAlbums(releases);
      });
  };

  const albumCards = albums.map(album => {
    return (
      <li key={album.id}>
        <AlbumCard 
          artist={artist} 
          album={album.title} 
          album_id={album.id} 
        />
      </li>
    );
  });

  return (
    <section className={styles.AlbumDeck}>
      <button onClick={() => handlePageChange(page - 1)}>Back</button>
      <h2>Albums by: {artist}</h2>
      <ul>
        {albumCards}
      </ul>
      <button onClick={() => handlePageChange(page + 1)}>Next</button>
    </section>
  );

}

export const MemoAlbumDeck = memo(AlbumDeck);
