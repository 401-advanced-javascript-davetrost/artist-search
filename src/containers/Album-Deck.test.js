import React from 'react';
import { shallow } from 'enzyme';
import { MemoAlbumDeck } from './Album-Deck';

import reactRouterMock from 'react-router-dom';
jest.mock('react-router-dom', () => ({}));
reactRouterMock.useParams = () => ({ artist: 'Radiohead', id: '1234' });

import useAlbumsMock from './use-albums';
jest.mock('./use-albums', () => ({}));

describe('AlbumDeck component', () => {

  it('renders the loading state', () => {
    useAlbumsMock.useAlbums = () => ({ 
      loading: true, 
      albums: []
    });
    const wrapper = shallow(<MemoAlbumDeck />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list of albums', () => {
    useAlbumsMock.useAlbums = () => ({ 
      loading: false, 
      albums: [
        { id: '1', title: 'The Bends', imgUrl: 'www.randomimage.com' },
        { id: '2', title: 'Amnesiac', imgUrl: 'www.randomimage.com' },
        { id: '3', title: 'Kid A', imgUrl: 'www.randomimage.com' },
      ]
    });
    const wrapper = shallow(<MemoAlbumDeck />);
    expect(wrapper).toMatchSnapshot();
  });
});
