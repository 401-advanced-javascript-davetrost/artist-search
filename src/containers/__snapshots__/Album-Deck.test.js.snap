// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDeck component renders a list of albums 1`] = `
<Fragment>
  <div
    className="AlbumDeck"
  >
    <h1>
      <span>
        Albums by:
      </span>
       
      Radiohead
    </h1>
    <section>
      <button
        onClick={[Function]}
      >
        Back
      </button>
      <ul>
        <li
          key="1"
        >
          <AlbumCard
            album="The Bends"
            album_id="1"
            artist="Radiohead"
            imgUrl="www.randomimage.com"
          />
        </li>
        <li
          key="2"
        >
          <AlbumCard
            album="Amnesiac"
            album_id="2"
            artist="Radiohead"
            imgUrl="www.randomimage.com"
          />
        </li>
        <li
          key="3"
        >
          <AlbumCard
            album="Kid A"
            album_id="3"
            artist="Radiohead"
            imgUrl="www.randomimage.com"
          />
        </li>
      </ul>
      <button
        onClick={[Function]}
      >
        Next
      </button>
    </section>
  </div>
</Fragment>
`;

exports[`AlbumDeck component renders the loading state 1`] = `
<Fragment>
  <Loading />
</Fragment>
`;
